server/src/main/java/io/crate/execution/engine/aggregation/impl/CountAggregation.java

```diff
@@ -276,4 +283,31 @@ public MutableLong removeFromAggregatedState(RamAccounting ramAccounting,
         }
         return null;
     }
+
+    @Nullable
+    @Override
+    public DocValueAggregator<?> getDocValueAggregator(List<Symbol> aggregationReferences,
+                                                       java.util.function.Function<List<String>, List<MappedFieldType>> getMappedFieldTypes,
+                                                       DocTableInfo table,
+                                                       List<Literal<?>> optionalParams) {
+        if (aggregationReferences.size() == 1 && aggregationReferences.get(0).valueType().id() == ObjectType.ID && table != null) {
+            var allNotNullSubColumns = new ArrayList<>(table.notNullColumns());
+            for (var notNullCol : allNotNullSubColumns) {
+                if (notNullCol.isChildOf(((Reference)aggregationReferences.get(0)).column())) {
+                    var fieldTypes = getMappedFieldTypes.apply(List.of(notNullCol.fqn()));
+                    var notNullColRef = table.getReference(notNullCol);
+                    if (fieldTypes == null || notNullColRef == null) {
+                        continue;
+                    }
+                    var subColDocValAggregator = getDocValueAggregator(
+                        List.of(notNullColRef.valueType()),
+                        fieldTypes,
+                        optionalParams
+                    );
+                    if (subColDocValAggregator != null) return subColDocValAggregator;
```

Comment:
```suggestion
                    if (subColDocValAggregator != null) {
                         return subColDocValAggregator;
                    }
```

I'm surprised that checkstyle doesn't complain about this.